tests/validation/tests/rke/common.py

```diff
@@ -385,13 +523,15 @@ def teardown(self):
                 'namespace', service_info['namespace'])
 
 
-def validate_k8s_service_images(nodes, expectedimagesdict):
+def validate_k8s_service_images(nodes, k8s_version, rke_client, kubectl):
```

Comment:
We call this method ~8 times in the upgrade tests. I suggest storing the images for each k8s version as you grab them, and looking to see if the images are already present before generating the list from `rke` again

Reply:
@bmdepesa : In the existing upgrade tests, we had the images stored in a dictionary for a K8s version by obtaining them as env variables. This was a change requested as part of refactoring.

Reply:
@soumyalj I understand. I'm saying to save the images after you build the list from rke -- if you need to get the images for the same k8s version in a subsequent, get them from the list you have already populated instead of running `rke config` again to get the same list.

Reply:
Modified build_expectedimages_dict() to build the dictionary for a specific K8s version only if it does not exist. It will read from the global dictionary once populated.